### PR TITLE
Turn off Node 6 execution handler deprecation warning for in-the-box tasks

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Description**:
We need to turn off Node 6 execution handler deprecation warning messages for in-the-box tasks

_Chnagelog_:
- Added `tasks-exception-list.json` - config file with the list of tasks IDs that are presented in the [azure-pipelines-tasks](https://github.com/microsoft/azure-pipelines-tasks/) repo and still using Node 6 execution handler
- Added method `getTaskExceptionList`
- Modified `Create` method of `HandlerFactory` to respect tasks exception list

<details>
  <summary>List of added tasks</summary>

| Task name                         | Version | Task ID                              |
| --------------------------------- | ------- | ------------------------------------ |
| ACRTask                           | 0.175.2 | 769D88CB-515B-4456-A045-D9A4E11C90E3 |
| AppCenterDistribute               | 1.194.0 | B832BEC5-8C27-4FEF-9FB8-6BEC8524AD8A |
| AppCenterDistribute               | 2.173.0 | B832BEC5-8C27-4FEF-9FB8-6BEC8524AD8A |
| AppCenterDistribute               | 3.195.0 | B832BEC5-8C27-4FEF-9FB8-6BEC8524AD8A |
| AppCenterTest                     | 1.152.3 | AD5CD22A-BE4E-48BB-ADCE-181A32432DA5 |
| AzureMonitorAlerts                | 0.175.2 | 1d876d40-9aa7-11e7-905d-f541cc882994 |
| ContainerStructureTest            | 0.179.0 | 39bc2c9b-55b7-4835-89cd-6cc699ef7220 |
| CopyFiles                         | 2.190.1 | 5BFB729A-A7C8-4A78-A7C3-8D717BB7C13C |
| DownloadFileshareArtifacts        | 1.154.2 | E3CF3806-AD30-4EC4-8F1E-8ECD98771AA0 |
| DownloadGithubNpmPackage          | 1.0.5   | 97411e3d-0241-4b1f-9607-2d2c04b4df51 |
| DownloadGitHubNugetPackage        | 1.0.8   | 9fac244b-8d7c-4d8e-a003-2097daa3270f |
| DownloadGitHubRelease             | 0.160.3 | 263abc27-4582-4174-8789-af599697778e |
| JenkinsDownloadArtifacts          | 1.190.0 | 86c37a92-59a7-444b-93c7-220fcf91e29c |
| npmAuthenticate                   | 0.179.0 | ad884ca2-732e-4b85-b2d3-ed71bcbd2788 |
| NuGetCommand                      | 2.194.0 | 333b11bd-d341-40d9-afcf-b32d5ce6f23b |
| NuGetInstaller                    | 0.179.0 | 333b11bd-d341-40d9-afcf-b32d5ce6f23b |
| NuGetPublisher                    | 0.179.0 | 333b11bd-d341-40d9-afcf-b32d5ce6f25b |
| NuGetRestore                      | 1.179.0 | 333b11bd-d341-40d9-afcf-b32d5ce6f23b |
| NuGetToolInstaller                | 0.179.1 | 2C65196A-54FD-4A02-9BE8-D9D1837B7C5D |
| NuGetToolInstaller                | 1.179.0 | 2C65196A-54FD-4A02-9BE8-D9D1837B7C5D |
| NuGet                             | 0.179.0 | 2661B7E5-00F9-4DE1-BA41-04E68D70B528 |
| OpenPolicyAgentInstaller          | 0.157.2 | 50817E39-E160-45E1-A825-1C746B7D2EB2 |
| PipAuthenticate                   | 0.179.0 | 5e3feff0-c5ae-11e8-a7d0-4bd3b8229800 |
| PipAuthenticate                   | 1.177.1 | 5e3feff0-c5ae-11e8-a7d0-4bd3b8229800 |
| PublishCodeCoverageResults        | 2.159.0 | 2A7EBC54-C13E-490E-81A5-D7561AB7CD97 |
| PublishTestResults                | 1.193.0 | 0B0F01ED-7DDE-43FF-9CBB-E48954DAF9B1 |
| PyPIPublisher                     | 0.167.2 | 2d8a1d60-8ccd-11e7-a792-11ac56e9f553 |
| TwineAuthenticate                 | 0.179.0 | e4d58330-c771-11e8-8f8f-81fbb42e2824 |
| TwineAuthenticate                 | 1.177.1 | e4d58330-c771-11e8-8f8f-81fbb42e2824 |
| UniversalPackages                 | 0.190.0 | e0b79640-8625-11e8-91be-db2878ff888a |
| VisualStudioTestPlatformInstaller | 1.151.3 | 2C65196A-54FD-4A02-9BE8-D9D1837B7111 |
| VSTest                            | 2.195.0 | EF087383-EE5E-42C7-9A53-AB56C98420F9 |
| XamarinTestCloud                  | 1.167.2 | 049918CB-1488-48EB-85E8-C318ECCAAA74 |

</details>

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:** 
- microsoft/build-task-team/issues/1878

**Checklist**:
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: src/Agent.Worker/Handlers/HandlerFactory.cs [hunk 1]
```diff
@@ -117,5 +127,23 @@ public sealed class HandlerFactory : AgentService, IHandlerFactory
             handler.TaskDirectory = taskDirectory;
             return handler;
         }
+
+        /// <summary> 
+        /// This method provides a list of pipeline tasks for which we don't want to display the warning about the Node6 execution handler.
```
Comment (anatolybolshakov) [inline src/Agent.Worker/Handlers/HandlerFactory.cs:132]: Nit: let's also mention that these are in-the-box tasks for visibility

Comment (alexander-smolyakov) [reply src/Agent.Worker/Handlers/HandlerFactory.cs]: Thanks, fixed!

**Review by EzzhevNikita — APPROVED** [review]

File: src/Agent.Worker/Handlers/HandlerFactory.cs [hunk 2]
```diff
@@ -117,5 +127,31 @@ public sealed class HandlerFactory : AgentService, IHandlerFactory
...
+
+            if (File.Exists(exceptionListFile))
+            {
+                try
+                {
+                    exceptionList = IOUtil.LoadObject<List<Guid>>(exceptionListFile);
+                }
+                catch (Exception ex)
+                {
+                    Trace.Info($"Unable to serialize exception list {ex}");
```
Comment (anatolybolshakov) [inline src/Agent.Worker/Handlers/HandlerFactory.cs:149]: Nit: this looks more like deserialization 😃

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

File: src/Agent.Worker/Handlers/HandlerFactory.cs (hunk 2 above)
Comment (alexander-smolyakov) [reply src/Agent.Worker/Handlers/HandlerFactory.cs]: Nice catch, fixed it.

**Review by max-zaytsev — APPROVED** [review]

**Pull request merged**